rules/prevent-abbreviations.js

```diff
@@ -0,0 +1,244 @@
+'use strict';
+const defaultsDeep = require('lodash.defaultsdeep');
+const toPairs = require('lodash.topairs');
+
+const getDocsUrl = require('./utils/get-docs-url');
+const avoidCapture = require('./utils/avoid-capture');
+
+const defaultReplacements = {
+	err: {
+		error: true
+	},
+	cb: {
+		callback: true
+	},
+	opts: {
+		options: true
+	},
+	str: {
+		string: true
+	},
+	obj: {
+		object: true
+	},
+	num: {
+		number: true
+	},
+	val: {
+		value: true
+	},
+	e: {
+		event: true,
+		error: true
+	},
+	evt: {
+		event: true
+	},
+	el: {
+		element: true
+	},
+	req: {
+		request: true
+	},
+	res: {
+		response: true,
+		result: true
+	},
+	btn: {
+		button: true
+	},
+	msg: {
+		message: true
+	},
+	len: {
+		length: true
+	},
+	env: {
+		environment: true
+	},
+	dev: {
+		development: true
+	},
+	prod: {
+		production: true
+	},
+	tmp: {
+		temporary: true
+	},
+	arg: {
+		argument: true
+	},
+	args: {
+		arguments: true
+	},
+	tbl: {
+		table: true
+	},
+	db: {
+		database: true
+	},
+	ctx: {
+		context: true
+	},
+	mod: {
+		module: true
+	}
+};
+
+const prepareOptions = ({extendDefaults = true, replacements = {}} = {}) => {
+	const mergedReplacements = extendDefaults ?
+		defaultsDeep({}, replacements, defaultReplacements) :
+		replacements;
+
+	return new Map(toPairs(mergedReplacements).map(([discouragedName, replacements]) => {
+		return [discouragedName, new Map(toPairs(replacements))];
+	}));
+};
+
+const getVariableReplacements = (replacements, variable) => {
+	const variableNameReplacements = replacements.get(variable.name);
+
+	if (!variableNameReplacements) {
+		return [];
+	}
+
+	return [...variableNameReplacements.keys()]
```

Comment:
Just curious, is `keys()` not an array here?

Reply:
No, it's an iterator: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/keys